Spigot-API-Patches/0005-Adventure.patch

```diff
@@ -0,0 +1,2059 @@
++
++    /**
++     * Sets the name displayed to entries for this team
++     *
++     * @param displayName New display name
++     * @throws IllegalArgumentException if displayName is longer than 128
++     *     characters.
++     * @throws IllegalStateException if this team has been unregistered
++     */
++    void displayName(@Nullable net.kyori.adventure.text.Component displayName) throws IllegalStateException, IllegalArgumentException;
++
++    /**
++     * Gets the prefix prepended to the display of entries on this team.
++     *
++     * @return Team prefix
++     * @throws IllegalStateException if this team has been unregistered
++     */
++    @NotNull net.kyori.adventure.text.Component prefix() throws IllegalStateException;
++
++    /**
++     * Sets the prefix prepended to the display of entries on this team.
++     *
++     * @param prefix New prefix
++     * @throws IllegalArgumentException if prefix is null
++     * @throws IllegalArgumentException if prefix is longer than 64
++     *     characters
++     * @throws IllegalStateException if this team has been unregistered
++     */
++    void prefix(@Nullable net.kyori.adventure.text.Component prefix) throws IllegalStateException, IllegalArgumentException;
++
++    /**
++     * Gets the suffix appended to the display of entries on this team.
++     *
++     * @return the team's current suffix
++     * @throws IllegalStateException if this team has been unregistered
++     */
++    @NotNull net.kyori.adventure.text.Component suffix() throws IllegalStateException;
++
++    /**
++     * Sets the suffix appended to the display of entries on this team.
++     *
++     * @param suffix the new suffix for this team.
++     * @throws IllegalArgumentException if suffix is null
++     * @throws IllegalArgumentException if suffix is longer than 64
++     *     characters
++     * @throws IllegalStateException if this team has been unregistered
++     */
++    void suffix(@Nullable net.kyori.adventure.text.Component suffix) throws IllegalStateException, IllegalArgumentException;
++
++    /**
++     * Gets the color of the team.
++     * <br>
++     * This only sets the team outline, other occurrences of colors such as in
++     * names are handled by prefixes / suffixes.
++     *
++     * @return team color, defaults to {@link ChatColor#RESET}
++     * @throws IllegalStateException if this team has been unregistered
++     */
++    @NotNull net.kyori.adventure.text.format.NamedTextColor color() throws IllegalStateException;
++
++    /**
++     * Sets the color of the team.
++     * <br>
++     * This only sets the team outline, other occurrences of colors such as in
++     * names are handled by prefixes / suffixes.
++     *
++     * @param color new color, must be non-null. Use {@link ChatColor#RESET} for
++     * no color
++     */
++    void color(@Nullable net.kyori.adventure.text.format.NamedTextColor color);
```

Comment:
I'd also make this `TextColor`, if it's not named either find nearest named or throw error.

Reply:
These abstractions need to last the test of time, it's not unreasonable to think teams may support hex colors in the future.

Reply:
This should not be `TextColor`; if they support hex in the future, a new overload can be added. This is an ABI-safe change in that case (assuming we keep the existing one); the other method however can return `TextColor` just fine as it is from the internal represented colour, and it must (as you note) pass the test of time.